velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -496,6 +496,22 @@ std::string HiveDataSink::stateString(State state) {
 void HiveDataSink::computePartitionAndBucketIds(const RowVectorPtr& input) {
   VELOX_CHECK(isPartitioned() || isBucketed());
   if (isPartitioned()) {
+    if (!hiveConfig_->allowNullPartitionKeys(
```

Comment:
I assume this check should be used only when partition keys are generated at runtime. For queries that write to "fixed" partitions, this check should happen much earlier before Velox execution even starts. It would be nice to clarify this in the doc.

Reply:
good advice, i have added these notes in the config doc.